libraries/chain/controller.cpp

```diff
@@ -4840,16 +4833,29 @@ struct controller_impl {
       return conf.block_validation_mode == validation_mode::LIGHT || conf.trusted_producers.count(producer);
    }
 
+   int32_t max_reversible_blocks_allowed() const {
+      if (conf.max_reversible_blocks == 0)
+         return std::numeric_limits<int32_t>::max();
+
+      return fork_db.apply<int32_t>(
+         [&](const fork_database_legacy_t& forkdb) {
+            return std::numeric_limits<int32_t>::max();
+         },
+         [&](const fork_database_if_t& forkdb) {
+            return conf.max_reversible_blocks - forkdb.size();
```

Comment:
`conf.max_reversible_blocks` is a `uint32_t`.

What if a user provides a value greater than 2^31? The difference may still be greater than 2^31 and not be representable as an `int32_t`.